qforce/qm/orca.py

```diff
@@ -0,0 +1,624 @@
+import os.path
+
+from colt import Colt
+import numpy as np
+from ase.units import Hartree, mol, kJ, Bohr
+#
+from .qm_base import WriteABC, ReadABC
+from ..elements import ATOM_SYM
+
+
+class Orca(Colt):
+    _user_input = """
+
+    charge_method = esp :: str :: [cm5, esp]
+    
+    # QM method to be used for geometry optimisation
+    opt_method = XTB2 :: str
+    # For high-throughput use XTB2
+    # For accuracy use r2SCAN-3c
+    
+    # QM method to be used for hessian calculation
+    # Note: The accuracy of this method determines the accuracy of bond, 
+    # angle and improper dihedral.
+    hess_method = B3LYP D4 def2-TZVP def2/J RIJCOSX
```

Comment:
not sure if I like that all methods and basis sets are given in one command. This can make _method arbitrarily dependent on the order of keywords. (same for other methods)

Reply:
I think this is a quite difficult problem. One solution would be having one keyword for the functional, one keyword for the basis set, one keyword for the dispersion correction, one keyword for the auxiliary basis set, which seems to be giving too many keywords.
If it is just about the order of the keywords, one might generate the key as `set(method.split())`?

Reply:
One reason is indeed the order, the other is the consistency with the other QM methods in Q-Force. But the latter is probably less important, so let's fix the former. I would suggest something like:
`method =  sort(method.split()).join(' ')`
so we end up with a sorted str

Reply:
@selimsami Thanks for the comment. I wonder where do you think such a method should be in?
I think that it should probably be here

https://github.com/selimsami/qforce/blob/e51178fe33f279a3bb08c1a3383ab59ecd67fc83/qforce/fragment.py#L234

where I change it to         `self.graph.graph['qm_method'] = normalise_methods(qm_method)`
I think I will do it when the test is on so I can be sure that it is doing what I think it is doing.

Reply:
Hmm I think would put this in ORCA only. The other software take care of method/basis set/dispersion and other settings separately.

Reply:
@selimsami This is also my initial thought but I'm not sure of where should I put it, as this information is in the Colt object.

Reply:
You could add your own colt validator:

```
from Colt.validator import Validator

Validator.add_validator("qm_method", normalize_qm_method)

def normalize_qm_method(method):
    return sort(method.split()).join(' ')
```

Then:

```
opt_method = XTB2 :: qm_method
hess_method = B3LYP D4 def2-TZVP def2/J RIJCOSX :: qm_method
...
```

Reply:
@selimsami Thank you for the suggestion. If this is the only issue, would you mind doing a PR on this after this one is merged? The pycolt don't have documentation so I'm not sure of how to use it.